envbuilder.go

```diff
@@ -556,23 +628,17 @@ func Run(ctx context.Context, opts options.Options) error {
 			return fmt.Errorf("restore mounts: %w", err)
 		}
 
-		// Create the magic file to indicate that this build
-		// has already been ran before!
-		file, err := opts.Filesystem.Create(magicDir.Built())
-		if err != nil {
-			return fmt.Errorf("create magic file: %w", err)
-		}
-		_ = file.Close()
```

Comment:
Review: Doing it later so it's applicable to both build and cached image. This might also fix an edge case (at least theoretically) where an error & restart could result in onCreateScript never being run.